src/selectors/entities/groups.ts

```diff
@@ -165,4 +166,13 @@ export const getAllAssociatedGroupsForReference = createSelector(
     (allGroups) => {
         return Object.entries(allGroups).filter((entry) => (entry[1].allow_reference && entry[1].delete_at === 0)).map((entry) => entry[1]);
     },
-);
\ No newline at end of file
+);
+
+export const getCurrentUserGroupMentionKeys = createSelector(
+    getAllAssociatedGroupsForReference,
+    (groups: Array<Group>) => {
+        const keys: UserMentionKey[] = [];
+        groups.forEach((group) => keys.push({key: `@${group.name}`}));
+        return keys;
+    },
+);
```

Comment:
Do we want to add a unit test for this selector specifically?

Reply:
Good idea! Added